src/m365/planner/commands/task/task-add.ts

```diff
@@ -0,0 +1,387 @@
+      args.options.assignedToUserIds.split(',').map(x => assignments[x] = {
+        "@odata.type": "microsoft.graph.plannerAssignment", "orderHint": " !"
+      });
+    }
+    return assignments;
+  }
+
+  private getBucketId(args: CommandArgs, planId: string): Promise<string> {
+    if (args.options.bucketId) {
+      return Promise.resolve(args.options.bucketId);
+    }
+
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/planner/plans/${planId}/buckets`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get<{ value: { id: string; name: string; }[] }>(requestOptions)
+      .then((response) => {
+        const bucket: { id: string; name: string; } | undefined = response.value.find(val => val.name === args.options.bucketName);
+    
+        if (!bucket) {
+          return Promise.reject(`The specified bucket does not exist`);
+        }
+    
+        return Promise.resolve(bucket.id);
+      });
+  }
+
+  private getPlanId(args: CommandArgs): Promise<string> {
+    if (args.options.planId) {
+      return Promise.resolve(args.options.planId);
+    }
+
+    return this
+      .getGroupId(args)
+      .then((groupId: string) => {
+        const requestOptions: any = {
+          url: `${this.resource}/v1.0/planner/plans?$filter=(owner eq '${groupId}')`,
+          headers: {
+            accept: 'application/json;odata.metadata=none'
+          },
+          responseType: 'json'
+        };
+
+        return request.get<{ value: { id: string; title: string; }[] }>(requestOptions);
+      }).then((response) => {
+        const plan: { id: string; title: string; } | undefined = response.value.find(val => val.title === args.options.planName);
+
+        if (!plan) {
+          return Promise.reject(`The specified plan does not exist`);
+        }
+
+        return Promise.resolve(plan.id);
+      });
+  }
+
+  private getGroupId(args: CommandArgs): Promise<string> {
+    if (args.options.ownerGroupId) {
+      return Promise.resolve(args.options.ownerGroupId);
+    }
+
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/groups?$filter=displayName eq '${encodeURIComponent(args.options.ownerGroupName as string)}'`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get<{ value: { id: string; }[] }>(requestOptions)
+      .then(response => {
+        const group: { id: string; } | undefined = response.value[0];
+
+        if (!group) {
+          return Promise.reject(`The specified owner group does not exist`);
+        }
+
+        return Promise.resolve(group.id);
+      });
+  }
+
+  private getUserIds(logger: Logger, users: string): Promise<string[]> {
+    const userArr: string[] = users.split(',').map(o => o.trim());
+    let promises: Promise<{ value: User[] }>[] = [];
```

Comment:
Instead of defining `promises` here as a `let` with a default empty array which we later reassign to another array, let's define promises in line 248 as a `const`.